instana/agent/host.py

```diff
@@ -115,7 +115,7 @@ def can_send(self):
             self._boot_pid = current_pid
             logger.debug("Fork detected; Handling like a pro...")
             self.handle_fork()
-            return False
+            return True
```

Comment:
The expected behavior of the `can_send` method has been well summarized on line `105`.
Now on this line `118`, that behavior seems to have been completely destroyed, because this claims that the tracer is in a state in which it can send, but there doesn't seem to be a single check on the state machine, furthermore the call on line 117 even resets the state machine.
A purpose of this `can_send` was to give the recorder a green light for recording spans,
without the high risk of quickly building up a huge queue of spans (1000 or more), and introducing latency in tracing or even worse running out of memory (ending up OOM killed). Now the tracer seems to violate [the specification](https://github.ibm.com/instana/technical-documentation/tree/master/tracing/specification#span-buffering), which says that `Tracers SHOULD drop spans if they cannot be sent to the agent after collecting 1000 spans or more.` Because previously there was no buffering at all, but now this adds potentially unlimited buffering. because there is nothing else to check and limit the queue_size.

Reply:
Got it. I will work on a better solution here then.